src/dragonball/src/api/v1/vmm_action.rs

```diff
@@ -488,7 +507,32 @@ impl VmmService {
         Ok(VmmData::Empty)
     }
 
+    /// Setup dragonball tracing.
+    fn setup_tracing(&self, trace_info: TraceInfo) -> VmmRequestResult {
+        let mut tracer = self.tracer.lock().unwrap();
+        tracer
+            .setup_tracing(trace_info)
+            .map(|_| VmmData::Empty)
+            .map_err(VmmActionError::TracingFailed)
```

Comment:
please separate this `TracingFailed` Error into 2 errors : `SetupTracingFailed` and `EndTracingFailed` so that we could know which operation goes wrong

Reply:
Done.